### DISCUSSION TITLE
How can I make the JS imported using jiti use the current project's package.json?

### DISCUSSION THREAD

**Discussion by EmptyDreams** [unjs/community#17] [Q&A]
I want to use `jiti` to load JS/TS files from any directory and I hope it can use the dependencies in my project.

For example, I have introduced `jiti` as a dependency. Now I have a JS file:

```javascript
import { createJiti } from 'jiti';
module.exports = { createJITI };
```

Now I use the following code to run this JS:

```typescript
const jiti = createJiti(__filename);
jiti.import('X://xx/xxx.js')
    .then(it => console.log(it));
```

However, an error occurs when running: `Error: Cannot find module 'jiti'`.

What should I do to run this JS file properly?

**Comment by pi0** [comment]
Hi. Would you be able to create a minimal reproduction?

**Reply by EmptyDreams** [reply]
> Hi. Would you be able to create a minimal reproduction?

[example.zip](https://github.com/user-attachments/files/16468011/example.zip)

If things go as expected, this code will throw an error during execution because "ex.js" cannot find the "jiti" module.
